pkg/validation/validation.go

```diff
@@ -0,0 +1,51 @@
+package validation
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const ErrorNotFoundAgent = "unsupported fence agent %s:"
+
+type validateAgentExistence struct {
+	isFileExist func(string) (fs.FileInfo, error)
+}
+
+type AgentValidator interface {
+	ValidateAgentName(agent string) (admission.Warnings, error)
+}
+
+func NewAgentValidator(fileGetter func(string) (fs.FileInfo, error)) AgentValidator {
```

Comment:
the validator interface should not care about files, that's just the default validation method.